internal/operator-controller/rukpak/render/validators/validator.go

```diff
@@ -86,3 +95,144 @@ func CheckPackageNameNotEmpty(rv1 *render.RegistryV1) []error {
+	}
+
+	for _, depl := range rv1.CSV.Spec.InstallStrategy.StrategySpec.DeploymentSpecs {
+		delete(webhooksByDeployment, depl.Name)
+	}
+
+	var errs []error
+	// Loop through sorted keys to keep error messages ordered by deployment name
+	for _, deploymentName := range slices.Sorted(maps.Keys(webhooksByDeployment)) {
+		webhookDefns := webhooksByDeployment[deploymentName]
+		slices.SortFunc(webhookDefns, func(a, b v1alpha1.WebhookDescription) int {
+			return cmp.Or(cmp.Compare(a.Type, b.Type), cmp.Compare(a.GenerateName, b.GenerateName))
+		})
+		for _, webhookDef := range webhookDefns {
+			errs = append(errs, fmt.Errorf("webhook '%s' of type '%s' references non-existent deployment '%s'", webhookDef.GenerateName, webhookDef.Type, webhookDef.DeploymentName))
+		}
+	}
+	return errs
+}
+
+// CheckWebhookNameUniqueness checks that each webhook definition of each type (validating, mutating, or conversion)
+// has a unique name. Webhooks of different types can have the same name. Errors are sorted by webhook type
+// and name.
+func CheckWebhookNameUniqueness(rv1 *render.RegistryV1) []error {
+	webhookNameSetByType := map[v1alpha1.WebhookAdmissionType]sets.Set[string]{}
+	duplicateWebhooksByType := map[v1alpha1.WebhookAdmissionType]sets.Set[string]{}
+	for _, wh := range rv1.CSV.Spec.WebhookDefinitions {
+		if _, ok := webhookNameSetByType[wh.Type]; !ok {
+			webhookNameSetByType[wh.Type] = sets.Set[string]{}
+		}
+		if webhookNameSetByType[wh.Type].Has(wh.GenerateName) {
+			if _, ok := duplicateWebhooksByType[wh.Type]; !ok {
+				duplicateWebhooksByType[wh.Type] = sets.Set[string]{}
+			}
+			duplicateWebhooksByType[wh.Type].Insert(wh.GenerateName)
+		}
+		webhookNameSetByType[wh.Type].Insert(wh.GenerateName)
+	}
+
+	var errs []error
+	for _, whType := range slices.Sorted(maps.Keys(duplicateWebhooksByType)) {
+		for _, webhookName := range slices.Sorted(slices.Values(duplicateWebhooksByType[whType].UnsortedList())) {
+			errs = append(errs, fmt.Errorf("duplicate webhook '%s' of type '%s'", webhookName, whType))
+		}
+	}
+	return errs
+}
+
+// CheckConversionWebhooksReferenceOwnedCRDs checks defined conversion webhooks reference bundle owned CRDs.
+// Errors are sorted by webhook name and CRD name.
+func CheckConversionWebhooksReferenceOwnedCRDs(rv1 *render.RegistryV1) []error {
+	//nolint:prealloc
+	var conversionWebhooks []v1alpha1.WebhookDescription
+	for _, wh := range rv1.CSV.Spec.WebhookDefinitions {
+		if wh.Type != v1alpha1.ConversionWebhook {
+			continue
+		}
+		conversionWebhooks = append(conversionWebhooks, wh)
+	}
+
+	if len(conversionWebhooks) == 0 {
+		return nil
+	}
+
+	ownedCRDNames := sets.Set[string]{}
+	for _, crd := range rv1.CSV.Spec.CustomResourceDefinitions.Owned {
+		ownedCRDNames.Insert(crd.Name)
+	}
+
+	slices.SortFunc(conversionWebhooks, func(a, b v1alpha1.WebhookDescription) int {
+		return cmp.Compare(a.GenerateName, b.GenerateName)
```

Comment:
Hi @perdasilva 

I am not finding where we are setting the GenerateName 
Can you help me out and link it ?

Reply:
It used to be [here](https://github.com/operator-framework/operator-controller/pull/1914/commits/7861571eb73dbda49b27b0c1e0f8c193d5870202#diff-867fc86b4cf60bb0357a2d94f2c61a76d3cf9f0b52d2aa4b7b32f4941cded7c3L312)

Reply:
For the RBAC resources we create, we use [this](https://github.com/operator-framework/operator-controller/blob/main/internal/operator-controller/rukpak/render/render.go#L120) function, which uses [the original rukpak fn](https://github.com/operator-framework/operator-controller/blob/main/internal/operator-controller/rukpak/util/util.go#L13). This currently keeps names below 63 characters.

Reply:
If we will do == OLV0 them I think that would be right:

OLM v0: Webhooks utilize metadata.generateName from csv.spec.webhookdefinitions.generateName as a prefix. Corresponding services are named <generateName>-service.
- Example: Refer to [webhook.go#L91](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/webhook.go#L91) and [webhook.go#L254](https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/operators/webhook.go#L254).

Reply:
If we will do == OLV0 them I think that would be right:

OLM v0: Webhooks utilize metadata.generateName from csv.spec.webhookdefinitions.generateName as a prefix. Corresponding services are named <generateName>-service.
- Example:

https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/install/webhook.go#L91

https://github.com/operator-framework/operator-lifecycle-manager/blob/master/pkg/controller/install/webhook.go#L254

So are we doing the same?

Reply:
That's essentially what we had as the first pass and Joe thought maybe this would be something where we could break the rule of doing exactly what olmv1 does ([comment](https://github.com/operator-framework/operator-controller/pull/1914#discussion_r2064234391)).

The service name remains the same ([i.e.](https://github.com/operator-framework/operator-controller/pull/1914/files#diff-95654e48627996e345270b32ee6275d5210a580c2d627c29af690027fde9a6dfR66)